include/libtorrent/extensions.hpp

```diff
@@ -370,6 +370,10 @@ namespace libtorrent {
 		// hidden
 		virtual ~peer_plugin() {}
 
+		// This function is expected to return the name of
+		// the plugin.
+		virtual span<char const> type() const { return span<char const>(); }
```

Comment:
I think ``string_view`` might be a better return type. and to default construct one, you can just do ``{}``.